scripts/download-deps.sh

```diff
@@ -1,18 +1,38 @@
 #!/usr/bin/env bash
-set +x
 
-binpath=${PWD}/$(dirname "$0")/../bin
-version="v0.0.0-20180823001027-3dcf91f64f63"
-cmds="deepcopy-gen defaulter-gen lister-gen client-gen informer-gen"
+set -euo pipefail
 
-for name in ${cmds}; do
-    if [[ ! -f $binpath/$name ]]; then
-        GOBIN=$binpath go get k8s.io/code-generator/cmd/"$name"@$version
+code_generator_version=0.0.0-20180823001027-3dcf91f64f63
+controller_gen_version=0.1.9
+
+dirname=$(dirname "$0")
+binpath=$PWD/$dirname/../bin
+
+function ensure-binary-version() {
+    local bin_name=$1
+    local bin_version=$2
+    local download_location=$3
+
+    local target_name=${bin_name}-${bin_version}
+    local link_path=${binpath}/${bin_name}
+
+    [ -e "${link_path}" ] && rm "${link_path}"
```

Comment:
Wouldn't it cause issues with the set -e if the link path does not exist?

Reply:
Nope, it's part of a compound command. I double checked by deleting one of the symlinks under bin. It's recreated properly.